dev/com.ibm.ws.concurrent/src/com/ibm/ws/concurrent/internal/ManagedExecutorServiceImpl.java

```diff
@@ -630,62 +495,33 @@ public void shutdown() {
     }
 
     /** {@inheritDoc} */
+    @SuppressWarnings("unchecked")
     @Override
     public <T> Future<T> submit(Callable<T> task) {
         Map<String, String> execProps = getExecutionProperties(task);
 
         WSContextService contextSvc = AccessController.doPrivileged(contextSvcAccessor);
 
-        // TODO replace this temporary prototype code
-        if (usePolicyExecutor) {
-            PolicyTaskCallback callback = new TaskLifeCycleCallback(this, contextSvc.captureThreadContext(execProps));
-            return policyExecutor.submit(task, callback);
-        }
-
-        ExecutorService execSvc = getExecSvc();
-
-        SubmittedTask<T> taskToSubmit = new SubmittedTask<T>(this, task, contextSvc.captureThreadContext(execProps), null);
-        if (taskToSubmit.future.isCancelled())
-            throw new RejectedExecutionException(Tr.formatMessage(tc, "CWWKC1110.task.canceled", taskToSubmit.getName(), name));
-
-        taskToSubmit.future.set(execSvc.submit((Callable<T>) taskToSubmit));
-
-        if (futures.add(taskToSubmit.future) && ++futureCount % FUTURE_PURGE_INTERVAL == 0)
+        PolicyTaskCallback callback = new TaskLifeCycleCallback(this, contextSvc.captureThreadContext(execProps));
+        Future<T> future = policyExecutor.submit(task, callback);
+        if (futures.add(future) && ++futureCount % FUTURE_PURGE_INTERVAL == 0)
```

Comment:
I expect we will eventually want to rely on the tracking of futures by the policy executor and not do it here, but for now this works well as a temporary measure.